Can't compile master: use of unstable library feature `integer_sign_cast`

Looks like b9940f44e included the following change:

```diff
diff --git a/crates/xdv/src/lib.rs b/crates/xdv/src/lib.rs
index d15d7e09..a2a95785 100644
--- a/crates/xdv/src/lib.rs
+++ b/crates/xdv/src/lib.rs
@@ -1229,7 +1228,7 @@ impl<'a, T: XdvEvents> Cursor<'a, T> {
             return Err(InternalError::NeedMoreData);
         }
 
-        let rv = unsafe { mem::transmute::<u8, i8>(self.buf[0]) };
+        let rv = self.buf[0].cast_signed();
         self.buf = &self.buf[1..];
         self.offset += 1;
         Ok(rv)
```

Which makes the `xdv` crate fail to compile with the following error:
```
error[E0658]: use of unstable library feature `integer_sign_cast`
    --> crates/xdv/src/lib.rs:1231:30
     |
1231 |         let rv = self.buf[0].cast_signed();
     |                              ^^^^^^^^^^^
     |
     = note: see issue #125882 <https://github.com/rust-lang/rust/issues/125882> for more information
```

Compiling tectonic on the immediate parent commit, c161cb53, works fine.

I'm on Linux 6.12 and Rust 1.85.0.

--- a/crates/xdv/src/lib.rs
+++ b/crates/xdv/src/lib.rs
@@ -1228,7 +1228,8 @@ impl<'a, T: XdvEvents> Cursor<'a, T> {
             return Err(InternalError::NeedMoreData);
         }
 
-        let rv = self.buf[0].cast_signed();
+        // TODO: Once we're willing to bump MSRV to 1.87, switch to `cast_signed`
+        let rv = self.buf[0] as i8;
         self.buf = &self.buf[1..];
         self.offset += 1;
         Ok(rv)

